Eatery/Utils/Sort.swift

```diff
@@ -110,6 +110,56 @@ struct Sort {
         return eatery.sorted(by: sortByHoursClosure)
     }
 
+    // Merge Sort code inspired by https://github.com/raywenderlich/swift-algorithm-club/tree/master/Merge%20Sort
+
+    static func expandedMenuMergeSort(_ array: [ExpandedMenu.Item]) -> [ExpandedMenu.Item] {
```

Comment:
Haven't dug into it too deep, but what's the reasoning for creating your own sort function over using `array.sort` and giving a custom `isLessThan` closure?

Reply:
I assumed implementing merge sort would be more efficient than Swift's sort method.  However, I didn't do much research on the sort function when I did this.  Do you think using swift's sort method would be better?

Reply:
I don't have any references to back me up, but I think Swift's implementation is probably faster (and possibly more correct). I think Swift, like most languages, uses quicksort which can be significantly faster than mergesort due to avoiding an `O(n)` memory allocation

Reply:
https://developer.apple.com/documentation/swift/array/1688499-sort
https://swiftrocks.com/introsort-timsort-swifts-sorting-algorithm

I don't recall if we updated to Swift 5, in which case Swift would be using introsort which has merge sort's complexity but is unstable. If we did update, Swift 5+ is using Timsort which is stable. Stability also doesn't seem to be a problem in this case since I don't believe we care about the order of similarly priced items. I would usually stick to the sorting method that the programming language you're using resorts to unless you have a very good reason for deviating!